internal/tests/architecture/architecture_test.go

```diff
@@ -0,0 +1,138 @@
+package architecture
+
+import (
+	"testing"
+
+	"github.com/matthewmcnew/archtest"
+)
+
+func TestArchitectureDependencyViolationFor(t *testing.T) {
+	tests := []struct {
+		name                         string
+		basePackage                  string
+		ignoredDependencies          []string
+		notAllowedDependencies       []string
+		notAllowedDirectDependencies []string
+	}{
+		{
+			name:        "entities",
+			basePackage: "subscription-api/internal/entities",
+			notAllowedDependencies: []string{
+				"subscription-api/cmd/...",
+				"subscription-api/internal/clients/...",
+				"subscription-api/internal/controllers/...",
+				"subscription-api/internal/mailing/...",
+				"subscription-api/internal/db/...",
+				"subscription-api/internal/mocks/...",
+				"subscription-api/internal/services/...",
+				"subscription-api/internal/sql/...",
+				"subscription-api/internal/tests/...",
+				"subscription-api/internal/tests/...",
+			},
+			notAllowedDirectDependencies: []string{
```

Comment:
Why did you decide to use the approach of describing what it shouldn't depend on, instead of explicitly describing dependencies?

Reply:
Architecture tests are written to check for violation of dependencies. 
IMO it is useless to check that package has specific dependencies, because list of these dependencies will grow as project develops. 

Also the recommended library does not support functionality of approach described by you.

Reply:
> because the recommended by lector library does not support functionality of approach described by you.

Yes, you are right. That is why I chose another library. The example of tests shown by the lecturer and the capabilities of the library he proposed did not match. That is why I was wondering why you decided to choose a different approach than the one the lecturer mentioned. Just curious)

Reply:
OK, your point makes more sense than my despotic approach via forbiding. 

I'll left dependency tests in this PR as it is, but will add dependency tests in [next PR](https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-boredgus/pull/7) due to your recommendation.